crates/matrix-sdk-crypto/src/backups/mod.rs

```diff
@@ -241,43 +241,43 @@ impl BackupMachine {
 
         if let Some(user_signatures) = signatures.get(self.account.user_id()) {
             for device_key_id in user_signatures.keys() {
-                if device_key_id.algorithm() == DeviceKeyAlgorithm::Ed25519 {
```

Comment:
Does the `No need to check our own device here` comment apply regardless of the algorithm? I.e. should the `device_key_id.device_id() == self.account.device_id()` check be part of the new function or move to before the fn call?

Reply:
I would keep it as part of the algorithm, so we don't end up in a scenario where we accidentally do something with an unknown algorithm and attackers gain the ability to spoof algorithms trying to get us to mess up.

It does look like it would be fine either way but I prefer them to be completely separate codepaths.